### DISCUSSION TITLE
Calling REST API after signing in with OIDC "No refresh token found for authenticated user"

### DISCUSSION THREAD

**Discussion by Hannnes1** [Budibase/budibase#12481] [Help]
I am in the process of creating a custom OIDC provider to be able to sign in to Firebase Auth from Budibase (The OIDC provider will sit between the two).

I have come to the point that signing in seems to work. Now I tried to call a REST API with the authorization header set to `{{ Current User.OAuthToken }}`, but as soon as I make the API call I get the error "No refresh token found for authenticated user". The API call is still made, but the authorization header is `undefined`.

It is possible that there is some configuration that is wrong on my part, but without more information about the error / warning it is difficult to debug the issue.

What does this error mean, and what could the cause be?

**Comment by melohagan** [comment]
Hey @Hannnes1 

The OAuthToken is the **access_token**. The API is indicating that you need a **refresh_token**, but we currently don't support that. 

There is a related feature request open here: https://github.com/Budibase/budibase/issues/8035

**Reply by Hannnes1** [reply]
Thanks for the reply. Does that mean OIDC can't be used for authenticating REST APIs?

**Reply by melohagan** [reply]
That's correct. Some APIs may allow you to use the access_token after you have logged into Budibase with OIDC, but you won't be able to authenticate / use a refresh_token directly against an API. 

For example you cannot currently log-in with Google SSO, and then use a REST API that needs to be authorized through a different OIDC account

**Reply by Hannnes1** [reply]
So I think I actually figured out what our issue was. The REST API (our API, actually) returned a 401 since the OIDC configuration wasn't fully working on our end (so not because the token had expired). That 401 reply caused Budibase to show the error "No refresh token found for authenticated user".

The API never expects a refresh token, so that error was a bit confusing.

**Reply by melohagan** [reply]
Ah yes I can see that the error could be misleading:

```ts
if (!providerType || !oauth2?.refreshToken) {
      throw new Error("No refresh token found for authenticated user")
    }
```

In this case presumably the *providerType* wasn't supplied because the config was incorrect.

**Reply by Hannnes1** [reply]
Sorry for opening this old discussion back up, but this wasn't fully solved as it turns out.
Where does `providerType` come from, and what could cause it to be undefined? If i interpret the source code correctly it's just an enum that should be either `oidc` or `google`.

**Reply by melohagan** [reply]
I thought you had resolved the issue Hannes, is your SSO configuration correct?
https://docs.budibase.com/docs/authentication-and-sso

**Reply by Hannnes1** [reply]
I thought so too. Will check the SSO configuration again
